Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -145,6 +159,25 @@ - (void)handleRemoteNotificationsRegistered:(NSNotification *)notification
   [self sendEventWithName:@"remoteNotificationsRegistered" body:notification.userInfo];
 }
 
+- (void)handleRegisterUserNotificationSettings:(NSNotification *)notification
+{
+  if (self.requestPermissionsResolveBlock == nil) return;
+
+  NSMutableDictionary* notificationTypes = [NSMutableDictionary dictionaryWithObjectsAndKeys:@NO, @"alert", @NO, @"badge", @NO, @"sound", nil];
```

Comment:
It would be cleaner to write this as:

```
NSDictionary *notificationTypes = @{
  @"alert": @(notificationSettings.types & UIUserNotificationTypeAlert),
  ...
};

```
(may need an extra BOOL cast)